src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/mappers/TripPatternForDateMapper.java

```diff
@@ -99,8 +99,8 @@ public TripPatternForDate map(Timetable timetable, LocalDate serviceDate) {
 
     return new TripPatternForDate(
       timetable.getPattern().getRoutingTripPattern(),
-      times,
-      frequencies,
+      List.copyOf(times),
```

Comment:
Why do you take a defensive copy here - because the immutable list have better performance? Would it make more sense to do it inside the constructor?